File: 프로그래머스/LV0/코딩테스트 입문/다음에 올 숫자/다음에 올 숫자.py
def solution(common):
    if common[1] - common[0] == common[2] - common[1]:
        answer = common[0] + (common[1] - common[0]) * len(common)
    else:
        answer = common[0] * (common[1]//common[0]) ** (len(common))
    return answer

File: 프로그래머스/LV0/코딩테스트 입문/다음에 올 숫자/다음에 올 숫자.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <math.h>
// common_len은 배열 common의 길이입니다.
int solution(int common[], size_t common_len) 
{
    if(common[1] - common[0] == common[2] - common[1]) return common[0] + (common[1] - common[0]) * common_len;
    else return common[0] * (int)pow(common[1] / common[0], common_len);
}